### ISSUE TITLE
Materials: Linked object overridden material don't have shininess

### ISSUE THREAD

**Issue by Anthony-Gaudino** [FreeCAD/FreeCAD#9980]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Version

0.20 (Release)

### Full version info

```shell
OS: macOS 14.5
Word size of FreeCAD: 64-bit
Version: 0.22.0dev.38079 (Git)
Build type: Release
Branch: main
Hash: de508e437d13e7eb35baffceea496fb9107888cb
Python 3.11.9, Qt 5.15.13, Coin 4.0.2, Vtk 9.2.6, OCC 7.7.2
Locale: C/Default (C)
Installed mods: 
  * Launcher
  * Nodes 0.1.36
  * InventorLoader 1.4.0
  * Assembly4.backup1719958896.85335 0.50.12 (Disabled)
  * Render.backup1719959316.120863 2024.6.5 (Disabled)
  * woodworking 0.21.2.33771
  * ThreadProfile.backup1719959316.6397839 1.90.0 (Disabled)
  * MeshRemodel.backup1719959315.871541 1.9.23 (Disabled)
  * Silk 0.1.5
  * Defeaturing 1.2.2
  * Pyramids-and-Polyhedrons
  * Curves.backup1719958901.528519 0.6.36 (Disabled)
  * fasteners.backup1719958901.808955 0.5.20 (Disabled)
  * nurbs
  * sheetmetal.backup1719959316.3711638 0.4.21 (Disabled)
  * A2plus.backup1719958896.309429 0.4.64a (Disabled)
  * Manipulator 1.5.7
  * lattice2 1.0.0
  * CurvedShapes 1.0.9
  * toSketch 1.0.1
  * Curves 0.6.40
  * Assembly4 0.50.15
  * Assembly3 0.12.2
```

### Subproject(s) affected?

None

### Problem description

When I create any object in FreeCad, if I import that object to another file as a linked object, then set it to `Override Material: true` and changes the material to for example `Metalized` in the `Appearance...` task, the object doesn't have any shininess.

Here's the linked object with the `Metalized` material applied:
<img width="275" alt="Screenshot 2023-07-23 at 19 46 29" src="https://github.com/FreeCAD/FreeCAD/assets/2439728/4d11c206-b176-470a-8c85-dd31181852ae">
It appears as a solid black object.

Here's the original object with the same material applied:
<img width="285" alt="Screenshot 2023-07-23 at 19 47 46" src="https://github.com/FreeCAD/FreeCAD/assets/2439728/ce080182-d003-4130-b76b-8f5b18a97d2f">

### Anything else?

With other materials, it seems it uses the default material shininess, so for example, if you select the `brass` material, the shininess is the same as the default material.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by grubertechnic** [comment]
It does not work on the master either.


```
OS: Debian GNU/Linux 11 (bullseye) (X-Cinnamon/lightdm-xsession)
Word size of FreeCAD: 64-bit
Version: 0.21.0.33636 (Git)
Build type: Release
Branch: master
Hash: a5e350ad469f4030b46baa6fd3d1770375f1ddd1
Python 3.9.2, Qt 5.15.2, Coin 4.0.0, Vtk 9.0.1, OCC 7.7.1
Locale: English/United States (en_US)
Installed mods: 
  * QuickMeasure 2022.10.28
  * CurvedShapes 1.0.4
  * Curves 0.6.12
```



But in general i am not happy with the property `Override Material`, since this is no material, only appearance. This should be changed because it is misleading.

**Comment by Anthony-Gaudino** [comment]
Maybe the issue is around here: https://github.com/FreeCAD/FreeCAD/blob/a3851d4701e66c73265fafb3d16ea0dcbbd09c19/src/Gui/ViewProviderLink.cpp

I can see references for transparence but no shininess.

**Comment by luzpaz** [comment]
Maybe something that overlaps with @davesrocketshop's Material revamp ?

**Comment by Anthony-Gaudino** [comment]
Looking at the code and testing on FreeCad, it seems that only `diffuseColor` and transparency are overriden in the link. All other appearance attributes are ignored.

**Comment by grubertechnic** [comment]
@luzpaz : not really. It is an issue in the name of the properties of the link object.

**Comment by davesrocketshop** [comment]
Unrelated to this, I've noticed in other places that appearance only uses diffuse colour and transparency, including in the material editor preview. This will be reviewed as part of the material overhaul, but that is a long term project. This should probably be fixed earlier.

**Comment by davesrocketshop** [comment]
Looking at the code, what's being linked is the App::Color object, which includes a color and transparency value, but not the other appearance properties. I have no idea how complicated a fix might be. I'll dig deeper tomorrow.

**Comment by Anthony-Gaudino** [comment]
Updated Full version info as it still happens on 0.22.0dev.
There's still problems assigning materials to linked objects without affecting the parent.

**Comment by luzpaz** [comment]
@davesrocketshop can you review the relevancy of this ticket when you have a spare moment? TIA

**Comment by davesrocketshop** [comment]
Well, it's a mixed bag. When I use the "Inspect Appearance..." and "Inspect Material..." tools, they're all set correctly. But they're not displaying correctly.

Here's what I'm seeing:

![Image](https://github.com/user-attachments/assets/1ed7954d-710d-4943-b52f-7ec8a2e652c4)

Both sphere's have the same setting. The one on the right is the linked object. The same issue occurs across files

**Comment by davesrocketshop** [comment]
Never mind. It's working correctly. "Override Material" should probably be changed to "Override Appearance", and I just saw where those settings live

![Image](https://github.com/user-attachments/assets/cd55eca7-adcb-497b-9488-0098640407d6)

I'd close this as no longer relavent

**Comment by luzpaz** [comment]
> "Override Material" should probably be changed to "Override Appearance", and I just saw where those settings live

@davesrocketshop are you making a PR ?  If so, then you should auto-close this ticket through said PR.

**Comment by maxwxyz** [comment]
Closing in favor of https://github.com/FreeCAD/FreeCAD/issues/19135
